torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1556,8 +1557,18 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
       //
       // See [Sync Streams].
       if (!avoidRecordStreams_) {
-        c10::cuda::CUDACachingAllocator::recordStream(
-            inputs[i].storage().data_ptr(), ncclStream);
+        // TODO: fix for sparse tensors
```

Comment:
don't we need this fixed, since we need to support sparse tensors here?

Reply:
old TODO that isn't relevant anymore